source/guide_fastapi.rst

```diff
@@ -0,0 +1,167 @@
+
+.. code-block:: python
+
+  #!/usr/bin/env python3.6
+  import fastapi
+  import uvicorn
+
+  api = fastapi.FastAPI()
+
+  @api.get('/')
+  def calculate():
+    answer = 24 + 18
+    return {'answer': answer}
+
+  # To test this app locally, uncomment this line:
+  # uvicorn.run(api, host="localhost", port=8000)
+
+
+
+Python Virtual Environment
+--------------------------
+To install the packages your application depends on, you want to use a virtual environment that is isolated from the system python's packages.
+FastAPI requires Python 3.6+ so we are going to use the latest working version, which is Python 3.8 (due to a `dependency issue`_):
+
+::
+
+  [isabell@stardust ~]$ cd ~/fastapi
+  [isabell@stardust fastapi]$ python3.8 -m venv .env
+  [isabell@stardust fastapi]$ source .env/bin/activate
+  (.env) [isabell@stardust fastapi]$
+
+In your virtual environment, install the dependencies we are using in ``main.py``:
+
+::
+
+  (.env) [isabell@stardust fastapi]$ pip install fastapi uvicorn
+  (.env) [isabell@stardust fastapi]$
+
+.. note:: You can already check if your application is working by uncommenting the ``ucivorn.run()`` line and executing ``python main.py`` in the virtual environment. Open a second ssh connection and get the output with ``curl localhost:8000``
+
+While uvicorn_ handles the asynchronous side of FastAPI, we want gunicorn_ as a well-tested and production-ready solution to run and manage multiple uvicorn workers. Install it with two dependencies in your virtual environment:
+
+::
+
+  (.env) [isabell@stardust fastapi]$ pip install gunicorn uvloop httptools
+  (.env) [isabell@stardust fastapi]$
+
+You can deactivate the virtual environment context like this:
+
+::
+
+  (.env) [isabell@stardust fastapi]$ deactivate
+  [isabell@stardust fastapi]$
+
+
+
+Configuration
+=============
+
+Incoming https requests are routed through :manual:`nginx <background-http-stack>` to a :manual:`web backend <web-backends>` where our application server is listening.
+Let's get this working:
+
+Gunicorn
+--------
+First, we are telling gunicorn to spin up several uvicorn processes and listen on port 8000 for incoming requests.
+
+.. code-block:: python
+
+  import os
+  app_path = os.environ['HOME'] + '/fastapi'
+
+  # Gunicorn configuration
+  wsgi_app = 'main:api'
+  bind = ':8000'
+  chdir = app_path
+  workers = 4
+  worker_class = 'uvicorn.workers.UvicornWorker'
+  errorlog =  app_path + '/errors.log'
+
+
+
+This is a minimal working example based on gunicorn's `example config`_, you can find a comprehensive `list of options here`_.
+
+.. note:: Test if everything is working with ``~/fastapi/.env/bin/gunicorn --config ~/fastapi/conf.py``
+
+
+Supervisord
+-----------
+Next, create a configuration for supervisord in ``~/etc/services.d/fastapi.ini``:
+
+.. code-block::
+
+  directory=%(ENV_HOME)s/fastapi
+  command=%(ENV_HOME)s/fastapi/.env/bin/gunicorn --config %(ENV_HOME)s/fastapi/conf.py
```

Comment:
```
[phpbb1@antila fastapi]$ supervisorctl reread
ERROR: CANT_REREAD: File contains no section headers.
file: /home/phpbb1/etc/services.d/fastapi.ini, line: 1
'directory=%(ENV_HOME)s/fastapi\n'
```
This seems to miss the `[program:]` line. How about this?
```suggestion
  [program:fastapi]
  directory=%(ENV_HOME)s/fastapi
  command=%(ENV_HOME)s/fastapi/.env/bin/gunicorn --config %(ENV_HOME)s/fastapi/conf.py
```

Reply:
Yes. Good catch!